pkg/runner/run_context.go

```diff
@@ -50,9 +50,44 @@ func (rc *RunContext) String() string {
 	return fmt.Sprintf("%s/%s", rc.Run.Workflow.Name, rc.Name)
 }
 
+type stepStatus int
```

Comment:
What is the reason for the `stepStatus` being an `int`? To me (I'm no golang expert) it seems overly complicated to implement the marshaling interface here when a simple string would suffice?

Reply:
At first, I've used a simple string. However, having an enum type makes sense because you don't have to depend on the strings themselves. It should become easier to reuse, change or extend step statuses in future. For example:

- More statuses could be added like "cancelled" or "skipped" if needed, and there wouldn't be a need to change the already existing business logic.
- It's easier to rename them by taking advantage of the integrated code editor refactor features.
- It's easier to rename the strings themselves, like changing "success" to "successful" or something like that.
- Performance. String comparisons are usually not as efficient as simply comparing enums, etc.

I'm not A Go expert, so for me, it's more like a force of habit using enum types in cases like that.

Reply:
Thanks for the explanation, that makes sense to me. 👍